lib/project_types/extension/models/specification_handlers/checkout_ui_extension.rb

```diff
@@ -22,6 +33,78 @@ def build_resource_url(context:, shop:)
           return unless product
           format("/cart/%<variant_id>d:%<quantity>d", variant_id: product.variant_id, quantity: 1)
         end
+
+        private
+
+        def localization(context)
+          Dir.chdir(context.root) do
+            locale_filenames = Dir["locales/*"].select { |filename| valid_l10n_file?(filename) }
+            # Localization is optional
+            return {} if locale_filenames.empty?
+
+            total_size = locale_filenames.sum { |filename| File.size(filename) }
+            if total_size > L10N_BUNDLE_SIZE_LIMIT
+              raise Extension::Errors::BundleTooLargeError,
+                "Total size of all locale files must be less than #{CLI::Kit::Util.to_filesize(L10N_BUNDLE_SIZE_LIMIT)}"
+            end
+
+            default_locale_matches = locale_filenames.grep(/default/)
+            if default_locale_matches.size != 1
+              raise Extension::Errors::SingleDefaultLocaleError,
+                "There must be one and only one locale identified as the default locale, e.g. `en.default.json`"
+            end
+            default_locale = File.basename(File.basename(default_locale_matches.first, ".json"), ".default")
+
+            locale_filenames.map do |filename|
+              locale = File.basename(File.basename(filename, ".json"), ".default")
+              [locale.to_sym, Base64.strict_encode64(File.read(filename, mode: "rt", encoding: "UTF-8"))]
+            end
+              .yield_self do |encoded_files_by_locale|
+              {
+                localization: {
+                  default_locale: default_locale,
+                  translations: encoded_files_by_locale.to_h,
+                },
+              }
+            end
+          end
+        end
+
+        def valid_l10n_file?(filename)
+          return false unless File.file?(filename)
+
+          dirname = File.dirname(filename)
+          return false unless dirname == "locales"
+
+          ext = File.extname(filename)
+          if ext != ".json"
+            raise Extension::Errors::InvalidFilenameError,
+              "Invalid locale filename: `#{filename}`; Only .json files are allowed"
+          end
+
+          basename = File.basename(File.basename(filename, ".json"), ".default")
+          unless valid_locale_code?(basename)
+            raise Extension::Errors::InvalidFilenameError,
+              "Invalid locale filename: `#{filename}`; Locale code should be 2 or 3 letters, optionally "\
+              "followed by a two-letter region code, e.g. `fr-ca`"
```

Comment:
We should be promoting usage of a capitalized region code. Therefore, please replace this with `fr-CA`.